src/main/java/org/folio/circulation/domain/validation/RequestLoanValidator.java

```diff
@@ -26,15 +23,13 @@ public RequestLoanValidator(LoanRepository loanRepository) {
     final Request request = requestAndRelatedRecords.getRequest();
 
     return loanRepository.findOpenLoanForRequest(request).thenApply(loanResult -> loanResult
-      .failWhen(loan -> of(() -> loan != null && loan.getUserId().equals(request.getUserId())), loan -> {
-        Map<String, String> parameters = new HashMap<>();
-        parameters.put("itemId", request.getItemId());
-        parameters.put("userId", request.getUserId());
-        parameters.put("loanId", loan.getId());
-
-        String message = "This requester currently has this item on loan.";
-
-        return singleValidationError(new ValidationError(message, parameters));
-      }).map(loan -> requestAndRelatedRecords));
+        .failWhen(loan -> of(() -> loan != null && loan.getUserId().equals(request.getUserId())),
+            loan -> singleValidationError(
+                "ui-requests.mod-circulation.requesterHasItemOnLoan",
```

Comment:
One of the primary concerns around introducing codes into the validation errors produced by the back end APIs, that the client will the rely on, is that in effect those codes become a part of the contract between the module and the clients.

Once a client goes beyond presenting the errors as is, this leaves me wanting to explore questions like:
* How does a client know what validation errors are produced?
* How does a client know what parameters it can expect to be included with a particular error?
* How do we avoid the keys conflicting across the system?
* Are errors part of the interface or implementation?

Specifically regarding the codes presented in this change, they seem to include both the name of the expected client and the back end implementation. This suggests to me that folks are comfortable coupling these errors to both the client and server implementations.

I acknowledge that we've already created some of this implicit coupling as folks have already started using codes 
and there are already places where we rely on the presence of certain parameters (e.g. the loan policy ID IIRC) for certain errors in order to present other information.

My preference would be that would establish some kind of guidance for this before we get too deep into implementing this. If we don't, I anticipate that at some point we will need to collect, review and likely change many of the codes (and hence clients as well). 

All of that said, I think I acknowledged on some recent override work that FOLIO seems keen to implement them as needed on a case by case basis without that guidance in place. (cc: @alexanderkurash )

Reply:
I've created https://issues.folio.org/browse/STRIPES-750 "DECISION: Which module should host back-end translation files?" for this decision.